### PR TITLE
944 upgrade async graqphl and actix

### LINKED ISSUE #944: No response for shipments / requests ( intermittent )
The API on the demo server is at times hanging, and no response is returned to the client.
This is for either shipments or requisitions, and when it happens other queries are returning as normal.
For shipments, removing the `item` node makes the query run normally.

Linked issue: #938 

See also [Josh's comment](https://github.com/openmsupply/openmsupply-client/issues/1055#issuecomment-1068918594) about multiple requests in another issue.

### PR DESCRIPTION
**Pull request by andreievg**
closes #944

### CODE REVIEW

File: graphql/core/src/test_helpers.rs [hunk 1]
```diff
@@ -54,15 +58,22 @@ pub async fn run_test_gql_query<
 
     let mut app = actix_web::test::init_service(
         actix_web::App::new()
-            .data(connection_manager_data.clone())
-            .data(loader_registry.clone())
```
Comment (andreievg) [inline graphql/core/src/test_helpers.rs:58]: no reason to put connection manager and loader registry in app data (only in graphql data), as far as i can see

File: server/src/lib.rs [hunk 2]
```diff
@@ -63,7 +63,6 @@ pub async fn start_server(
 
     let mut http_server = HttpServer::new(move || {
         App::new()
-            .app_data(connection_manager_data_app.clone())
```
Comment (andreievg) [inline server/src/lib.rs:66]: no reason to put connection manager and loader registry in app data (only in graphql data), as far as i can see

File: graphql/lib.rs [hunk 3]
```diff
@@ -93,40 +93,36 @@ pub fn config(
...
-                .route("", actix_web::web::get().to(graphql)),
-        );
+        cfg.app_data(Data::new(schema))
+            .service(web::resource("/graphql").guard(guard::Post()).to(
+                |schema: Data<Schema>, http_req, req: GraphQLRequest| {
+                    graphql(schema, http_req, req)
+                },
+            ))
+            .service(
+                web::resource("/playground")
```
Comment (clemens-msupply) [inline graphql/lib.rs:103]: Isn't it possible to have it on the same url? "/playground" sounds a bit work in progress...

**Review by clemens-msupply — APPROVED** [review]
Looks good!

File: graphql/lib.rs (hunk 3 above)
Comment (andreievg) [reply graphql/lib.rs]: I forgot to test playground, thanks: b497237384f4218c3a09b3cbb98485118a970034

**Pull request merged**